tests/codegen/repr/transparent-opaque-ptr.rs

```diff
@@ -0,0 +1,111 @@
+//@ revisions: aarch64-linux aarch64-darwin
+//@ compile-flags: -O -C no-prepopulate-passes
+
+//@[aarch64-linux] compile-flags: --target aarch64-unknown-linux-gnu
+//@[aarch64-linux] needs-llvm-components: aarch64
+//@[aarch64-darwin] compile-flags: --target aarch64-apple-darwin
+//@[aarch64-darwin] needs-llvm-components: aarch64
+
+// See ./transparent.rs
+// Some platforms pass large aggregates using immediate arrays in LLVMIR
+// Other platforms pass large aggregates using by-value struct pointer in LLVMIR
+// Yet more platforms pass large aggregates using opaque pointer in LLVMIR
+// This covers the "opaque pointer" case.
+
+#![feature(no_core, lang_items, transparent_unions)]
+#![crate_type = "lib"]
+#![no_std]
+#![no_core]
+
+#[lang = "sized"]
+trait Sized {}
+#[lang = "freeze"]
+trait Freeze {}
+#[lang = "copy"]
+trait Copy {}
```

Comment:
Remark: having to write this every time seems a bit unfortunate, cc https://github.com/rust-lang/rust/issues/130375, though I'm not exactly sure what the concrete design of pulling out the `#[no_core]` stuffs would look like, would appreciate any advice.

Reply:
`//@ aux-build: minicore.rs`?

Reply:
... right. That might need to be a separate compiletest flag so that we don't look for a minicore.rs inside a `auxiliary` folder relative to the test.